### ISSUE TITLE
Can't compile widgetset for Vaadin 7.4.x

### ISSUE THREAD

**Issue by Alan47** [johndevs/gradle-vaadin-plugin#159]
Hello,

I tried upgrading my project to Vaadin 7.4.0 recently. The Gradle Vaadin Plugin is a real time-saver and makes working with Vaadin very easy, but unfortunately the "vaadinCompileWidgetset" target does not seem to work for Vaadin 7.4.x yet. The process does not fail, produces no errors or warnings, and it does run for a while, but it produces no output to speak of. Also, the "cleanVaadinCompileWidgetSet" target does not clean the widgetset directory. It just seems to do nothing at all.

I'm not sure if the new Vaadin version is already supported, given that it's just a few days old, but I thought that I should let you know that apparently some things work differently in the new version, which causes problems in your Gradle Plugin. For the time being, I down-graded Vaadin to 7.3.10 again, where your Plugin works like a charm.

Kind regards,

Alan

**Comment by christoph-frick** [comment]
dup of #153 ?

**Comment by Alan47** [comment]
Oh... yeah sorry I must have overlooked that one. My bad :(

**Comment by johndevs** [comment]
Yes, the widgetset compilation should again work with the latest plugin and Vaadin 7.4. Ensure you are using version 0.9.7 of the plugin in your build.

Created #160 for improving the logging of the widgetset compile.

**Comment by Alan47** [comment]
Hello again,

today I found the time to apply the advise from #153 to my project. Unfortunately the WidgetSet build still fails when I use Vaadin 7.4.0 (with gradle vaadin plugin 0.9.7). After a lot of fiddling with the logging parameters, I finally got gradle to display what's going on:

[ERROR] Unexpected internal compiler error
java.lang.IncompatibleClassChangeError: class com.google.gwt.dev.javac.BytecodeSignatureMaker$CompileDependencyVisitor has interface org.objectweb.asm.ClassVisitor as super class
at java.lang.ClassLoader.defineClass1(Native Method)
at java.lang.ClassLoader.defineClass(ClassLoader.java:760)
at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
at java.net.URLClassLoader.defineClass(URLClassLoader.java:455)
at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
at java.net.URLClassLoader$1.run(URLClassLoader.java:367)
at java.net.URLClassLoader$1.run(URLClassLoader.java:361)
at java.security.AccessController.doPrivileged(Native Method)
at java.net.URLClassLoader.findClass(URLClassLoader.java:360)
at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308)
at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
at com.google.gwt.dev.javac.BytecodeSignatureMaker.visitCompileDependenciesInBytecode(BytecodeSignatureMaker.java:227)
at com.google.gwt.dev.javac.BytecodeSignatureMaker.getCompileDependencySignature(BytecodeSignatureMaker.java:209)
at com.google.gwt.dev.javac.CompiledClass.getSignatureHash(CompiledClass.java:166)
at com.google.gwt.dev.javac.Dependencies$Ref.<init>(Dependencies.java:42)
at com.google.gwt.dev.javac.Dependencies$Ref.<init>(Dependencies.java:37)
at com.google.gwt.dev.javac.Dependencies.resolve(Dependencies.java:114)
at com.google.gwt.dev.javac.CompilationStateBuilder$CompileMoreLater.compile(CompilationStateBuilder.java:366)
at com.google.gwt.dev.javac.CompilationStateBuilder.doBuildFrom(CompilationStateBuilder.java:580)
at com.google.gwt.dev.javac.CompilationStateBuilder.buildFrom(CompilationStateBuilder.java:513)
at com.google.gwt.dev.javac.CompilationStateBuilder.buildFrom(CompilationStateBuilder.java:499)
at com.google.gwt.dev.cfg.ModuleDef.getCompilationState(ModuleDef.java:668)
at com.google.gwt.dev.Precompile.precompile(Precompile.java:255)
at com.google.gwt.dev.Precompile.precompile(Precompile.java:229)
at com.google.gwt.dev.Precompile.precompile(Precompile.java:145)
at com.google.gwt.dev.Compiler.run(Compiler.java:206)
at com.google.gwt.dev.Compiler.run(Compiler.java:158)
at com.google.gwt.dev.Compiler$1.run(Compiler.java:120)
at com.google.gwt.dev.CompileTaskRunner.doRun(CompileTaskRunner.java:55)
at com.google.gwt.dev.CompileTaskRunner.runWithAppropriateLogger(CompileTaskRunner.java:50)
at com.google.gwt.dev.Compiler.main(Compiler.java:127)
Shutting down PersistentUnitCache thread


Any hint on what might be going on here?

**Comment by Alan47** [comment]
Okay I did a full clean build of the project, that fixed the issue (see: http://stackoverflow.com/questions/23176110/gwt-compilation-error-incompatibleclasschangeerror).

The build is not producing any more errors. Instead, gradle always "skips" the vaadinCompileWidgetSet task, even if I run "cleanVaadinCompileWidgetSet" directly before that command. And still no widgetset files are being produced :(

**Comment by Alan47** [comment]
I forced the rebuild of the widgetset by using --rerun-tasks as gradle parameter. The glorious result is that the stack trace from above appears again.

**Comment by christoph-frick** [comment]
@Alan47 is there a `build/logs/widgetset-compile.log` with further clues?

also check https://vaadin.com/forum#!/thread/9316748 for clues.

**Comment by Alan47** [comment]
Hi,
unfortunately I don't have such a folder. I've seen it in your explanations in #153, but I can't find it anywhere in my project.

I think that the real issue is that my gradle still runs "asm 3.2" (see: https://groups.google.com/forum/#!msg/google-web-toolkit/_zaQy_Rh67c/9YLEND-88osJ).

Unfortunately, asm 5 uses a different namespace ("org.ow2.asm:asm-all:5.0.3") and I have no clue how to tell gradle that it should use "org.ow2.asm:asm-all:5.0.3" instead of "asm:asm:3.2"...

Or could there be another reason why it doesn't work?

**Comment by Alan47** [comment]
It appears that "providedCompile('org.ow2.asm:asm-all:5.0.3')" does the trick. I'm currently building the WidgetSet...

**Comment by christoph-frick** [comment]
@Alan47 first you should be sure, where it comes from. do a gradle `gradle dependencies` to see what is going on.  you will see some some package pulling in the wromg asm version.  an exclude on that import should make it go away (but might introduce errors with the package originally pulling that one in)

**Comment by Alan47** [comment]
@christoph-frick Thanks a lot for your help, it's much appreciated!

I've been FINALLY able to build the widgetset for Vaadin 7.4.0. Basically all that was "wrong" was the asm version. I used "gradle -q org.example.mywebapp:dependencyInsight --dependency asm" to find out what's wrong. As it turned out, Hibernate was referencing asm 3.2. I added "providedCompile('org.ow2.asm:asm-all:5.0.3')" to my gradle file to resolve the issue, and it seems that everything is working fine now.

Thanks a lot to @christoph-frick  and @johndevs !
